### PR TITLE
Sa order books

### PR DESCRIPTION
**Pull request by shaqsinc22**
-Books are Now in Order with Newest on Top, Admin Panel Books Index.

### CODE REVIEW

**Review by Linda-Diamond-Robbins — COMMENTED** [review]
You rock Shaeel!

File: app/controllers/admin/books_controller.rb [hunk 1]
```diff
@@ -38,6 +38,16 @@ def destroy
     end
   end
 
+  def check_downloads
```
Comment (troyleach) [inline app/controllers/admin/books_controller.rb]: This looks like new code?
where is the test?

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -73,6 +73,13 @@ def download_pdf
     )
   end
 
+  def download
+    @book = Book.find(params[:id])
+    @book.book_downloads += 1
```
Comment (troyleach) [inline app/controllers/books_controller.rb]: looks like new code...
where is the test?

File: app/models/book.rb [hunk 3]
```diff
@@ -43,4 +43,13 @@ def authors_names
 end
 
 
+def check_downloads
```
Comment (troyleach) [inline app/models/book.rb]: looks like new code
where is the test?

File: app/views/books/show.html.erb [hunk 4]
```diff
@@ -5,7 +5,11 @@
   </div>
   <div class="col-xs-3">
      <%= image_tag "#{@book_information[:image]}", {class: "small_image" } %> 
-     <button class="btn btn-success right_justify_text"><%= link_to "Download #{@book.title}", "#{@book.url}", :target => "_blank", class: "black_text"%></button>
+     <% if @book.check_downloads == true %>
```
Comment (troyleach) [inline app/views/books/show.html.erb]: check this out...
try this:
```if @book.check_downloads```

File: config/routes.rb [hunk 5]
```diff
@@ -103,21 +103,22 @@
 	# get '/comment/:id' => 'comments#show'
 	post '/comments' => 'comments#create'
 
-	resources :books do
-		get :download_pdf
-		resources :authors do
-	    end
-	end
+  resources :books do
+    # get :download_pdf
```
Comment (troyleach) [inline config/routes.rb]: what is this?

File: db/migrate/20170519012557_change_book_downloads_in_book.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class ChangeBookDownloadsInBook < ActiveRecord::Migration[5.0]
+  def change
+    change_column_default :books, :book_downloads, 0
+  end
+end
```
Comment (troyleach) [inline db/migrate/20170519012557_change_book_downloads_in_book.rb]: please combine these two migrations
just cleans up the code
(remember rake db:migrate)

File: db/schema.rb [hunk 7]
```diff
@@ -51,6 +51,8 @@
     t.string   "ebook_content_type"
     t.integer  "ebook_file_size"
     t.datetime "ebook_updated_at"
+    t.integer  "book_downloads",     default: 0
+    t.integer  "max_downloads"
   end
 
   create_table "categories", force: :cascade do |t|
```
Comment (troyleach) [inline db/schema.rb]: I up loaded a new book and this is where I got rerouted to

screen shot 2017-05-22 at 7 53 38 pm

**Review by troyleach — CHANGES_REQUESTED** [review]

File: app/controllers/books_controller.rb [hunk 8]
```diff
@@ -73,6 +73,13 @@ def download_pdf
     )
   end
 
+  def download
```
Comment (troyleach) [inline app/controllers/books_controller.rb]: what are we down loading here? <naming>

**Review by troyleach — CHANGES_REQUESTED** [review]

**Pull request merged**